### PR TITLE
Add docs for DotProduct Abs Reverse HardShrink SelectTable Scale HingeEmbeddingCriterion ELU Initializers and Adadelta

### PR DESCRIPTION
**Pull request by zhichao-li**
## What changes were proposed in this pull request?
Add docs for the following layers:
nn/DotProduct.md
nn/Abs.md
nn/Reverse.md
nn/HardShrink.md
nn/SelectTable.md
nn/Scale.md
nn/HingeEmbeddingCriterion.md
nn/ELU.md
optim/Adadelta.md

### CODE REVIEW

File: docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+**Scala:**
+```scala
+SelectTable[T](index: Int))
+```
+**Python:**
+```python
+SelectTable(self,
+                            dimension, bigdl_type="float")
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md]: adjust the indentation

File: docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+**Scala example:**
+
+```scala
+import com.intel.analytics.bigdl.utils._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+val input = T(Tensor[Double](2,3).randn(), Tensor[Double](2,1).randn())
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md]: use readable input.

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/ELU.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.utils._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.nn._
+
+val xs = Tensor[Float](4).randn()
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/ELU.md]: use readable input.

File: docs/docs/APIdocs/layers/nn/Merge Layers/Reverse.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+## Reverse ##
+
+ * Reverse the input w.r.t given dimension.
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Merge Layers/Reverse.md]: delete *

File: docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+## HingeEmbeddingCriterion ##
+
+Creates a criterion that measures the loss given an input `x` which is a 1-dimensional vector and a label `y` (`1` or `-1`).
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md]: Put this behind **Python**, also all other MD files.

File: docs/docs/APIdocs/layers/nn/Merge Layers/Abs.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+**Scala example:**
+```scala
+val abs = new Abs[Float]
+val input = Tensor[Float](2)
+input(1) = 21
+input(2) = -29
+print(abs.forward(input))
+```
+`output is:　21.0　29.0`
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Merge Layers/Abs.md]: Could you use a uniform style? Like


Output is
```
21.0
29.0
[com.intel.analytics.bigdl.tensor.DenseTensor of size 2]
```

File: docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+This is usually used for measuring whether two inputs are similar or dissimilar, e.g. using the L1 pairwise distance, and is typically used for learning nonlinear embeddings or semi-supervised learning.
+
+
+**Scala:**
+``` scala
+criterion = HingeEmbeddingCriterion([margin: Double = 1,
+                                       sizeAverage: Boolean = true])
```
Comment (SeaOfOcean) [inline docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md]: not need [ ] in arguments

File: docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+
+**Python example:**
+```python
+input = np.random.random(4)
+target = np.full(4, 1)
+print("input: " )
+print(input)
+print("target: ")
```
Comment (SeaOfOcean) [inline docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md]: maybe can use 
```
print "input: \n", input
```

File: docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md (hunk 1 above)
Comment (SeaOfOcean) [inline docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md]: remove self, 
scala arg name index not consistent with python dimension?

File: docs/docs/APIdocs/layers/nn/Merge Layers/Scale.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.utils._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.utils.{T, Table}
+
```
Comment (SeaOfOcean) [inline docs/docs/APIdocs/layers/nn/Merge Layers/Scale.md]: import Table is not used here

File: docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.utils.{T}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.utils.{T}
+
+val loss = HingeEmbeddingCriterion[Double](1, sizeAverage = false)
+val input = Tensor[Double](T(0.1, 2.0, 2.0, 2.0))
+println("input: \n" + input)
```
Comment (SeaOfOcean) [inline docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md]: use float

File: docs/docs/APIdocs/layers/nn/Merge Layers/DotProduct.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+```scala
+import com.intel.analytics.bigdl.utils._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+val mlp = DotProduct[Double]()
+val x = Tensor[Double](3).fill(1)
+val y = Tensor[Double](3).fill(2)
+println("input:")
```
Comment (SeaOfOcean) [inline docs/docs/APIdocs/layers/nn/Merge Layers/DotProduct.md]: Float

**Comment by SeaOfOcean** [comment]
use float as default

**Comment by zhichao-li** [comment]
cc @SeaOfOcean @qiuxin2012

File: docs/docs/APIdocs/layers/nn/Initializers/BilinearFiller.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+
+
+**Scala:**
+``` scala
+val initMethod = BilinearFiller
+
+```
+**Python:**
+```python
+init_method = BilinearFiller(bigdl_type="float")
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Initializers/BilinearFiller.md]: do we need bigdl_type? Most of time we don't expect user to use this way?

Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Initializers/BilinearFiller.md]: scala we didn't put [T] in constructor and we also don't expect user to use bigdl_type, right?

File: docs/docs/APIdocs/layers/nn/Initializers/BilinearFiller.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+
+val weightInitMethod = BilinearFiller
+val model = SpatialFullConvolution(2, 3, 2, 2).setName("sfconv")
+model.setInitMethod(weightInitMethod, Zeros)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Initializers/BilinearFiller.md]: Zeros is another initializer? Not clear enough here. Better write
```scala
val weightInitMethod = BilinearFiller
val biasInitMethod= Zeros
val model = SpatialFullConvolution(2, 3, 2, 2).setName("sfconv")
model.setInitMethod(weightInitMethod, biasInitMethod)
```

File: docs/docs/APIdocs/layers/nn/Initializers/ConstInitMethod.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+
+
+val weightInitMethod = ConstInitMethod(0.2)
+val biasInitMethod = ConstInitMethod(0.2)
+val model = Linear(3, 2).setName("linear1")
+model.setInitMethod(weightInitMethod, biasInitMethod)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Initializers/ConstInitMethod.md]: A little bit confusing here. You defined a layer but name it a model (usually models are defined as containers Sequential, etc.).  How about we write it this way: 
```scala
val layer = Linear(3, 2).setName("linear1")
layer.setInitMethod(weightInitMethod=ConstInitMethod(0.2), biasInitMethod=ConstInitMethod(0.2))
val model = Sequential()
model.add(layer)
println(model.getParametersTable().get("linear1").get)
```

File: docs/docs/APIdocs/layers/nn/Initializers/RandomNormal.md [hunk 15]
```diff
@@ -0,0 +1,71 @@
+## RandomNormal ##
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Initializers/RandomNormal.md]: title is RandomNormal but content is RandomUniform.

File: docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md [hunk 16]
```diff
@@ -0,0 +1,81 @@
+## HingeEmbeddingCriterion ##
+
+
+**Scala:**
+``` scala
+criterion = HingeEmbeddingCriterion(margin: Double = 1, sizeAverage: Boolean =true)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md]: this code will throw error when user copy/paste it on the scala REPL. Maybe we should make it this way? 
```scala
val criterion = HingeEmbeddingCriterion(margin = 1, sizeAverage =true)
```
This way it omitts the actual type of each argument. - Yet I think if user needs to know the actual type of the parameter, they can refer to ScalaDoc.

File: docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+```
+**Python:**
+```python
+HingeEmbeddingCriterion(margin=1, size_average=True, bigdl_type="float")
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md]: needs to be 
```python
m = HingeEmbeddingCriterion(margin=1, size_average=True, bigdl_type="float")
```

File: docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.utils.{T}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.utils.{T}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+
+val loss = HingeEmbeddingCriterion[Float](1, sizeAverage = false)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/HingeEmbeddingCriterion.md]: don't need [Float] if you use NumericFloat

File: docs/docs/APIdocs/layers/nn/Merge Layers/Abs.md [hunk 19]
```diff
@@ -0,0 +1,37 @@
+## Abs ##
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Merge Layers/Abs.md]: this should be put in Simple Layers.

Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Merge Layers/Abs.md]: merge layers mostly applied on table inputs.

Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Merge Layers/Abs.md]: you can just leave it there and I will confirm with Yiheng and helen about layer folders and adjust later.

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/ELU.md [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+plt.plot(xs, f(0.1), '-', label='fw ELU, alpha = 0.1')
+plt.plot(xs, f(1.0), '-', label='fw ELU, alpha = 0.1')
+plt.plot(xs, df(0.1), '-', label='dw ELU, alpha = 0.1')
+plt.plot(xs, df(1.0), '-', label='dw ELU, alpha = 0.1')
+
+plt.legend(loc='best', shadow=True, fancybox=True)
+plt.show()
+
+```
+![](ELU.png)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/ELU.md]: put pictures in the image folder. https://github.com/intel-analytics/BigDL/tree/master/docs/docs/Image

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/HardShrink.md [hunk 21]
```diff
@@ -0,0 +1,83 @@
+## HardShrink ##
+
+
+**Scala:**
+```scala
+HardShrink(lambda: Double = 0.5)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/HardShrink.md]: val m = ...

File: docs/docs/APIdocs/layers/optim/Adadelta.md [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+```
+**Python:**
+```python
+AdaDelta(decayrate = 0.9, epsilon = 1e-10, bigdl_type="float")
+```
+
+
+**Scala example:**
+```scala
+optimizer.setOptimMethod(new Adadelta(0.9, 1e-10))
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/optim/Adadelta.md]: Maybe too simple? You may take a look at https://github.com/intel-analytics/BigDL/blob/master/docs/docs/APIdocs/layers/optim/Adagrad.md

File: docs/docs/APIdocs/layers/optim/Adadelta.md [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+
+**Scala:**
+```scala
+Adadelta(decayRate: Double = 0.9, Epsilon: Double = 1e-10)
+```
+**Python:**
+```python
+AdaDelta(decayrate = 0.9, epsilon = 1e-10, bigdl_type="float")
+```
+
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/optim/Adadelta.md]: description should locate between instance creation and actual example.

File: docs/docs/APIdocs/layers/optim/Adadelta.md (hunk 22 above)
Comment (zhichao-li) [inline docs/docs/APIdocs/layers/optim/Adadelta.md]: Actually we would not expect user use ```Adadelta``` like this in https://github.com/intel-analytics/BigDL/blob/master/docs/docs/APIdocs/layers/optim/Adagrad.md . Simply create an Adadelta object and throw into an Optimizer that's the most straight forward or formal usage.

File: docs/docs/APIdocs/layers/nn/Initializers/RandomNormal.md (hunk 15 above)
Comment (zhichao-li) [inline docs/docs/APIdocs/layers/nn/Initializers/RandomNormal.md]: good catch.

File: docs/docs/APIdocs/layers/nn/Initializers/ConstInitMethod.md (hunk 14 above)
Comment (zhichao-li) [inline docs/docs/APIdocs/layers/nn/Initializers/ConstInitMethod.md]: How about just rename `model` to `linear1` as there's no need to introduce container here.

File: docs/docs/APIdocs/layers/optim/Adadelta.md (hunk 22 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/optim/Adadelta.md]: But we would not expect user to use `forward` and `backward` either, while we still put `forward`/`backward` in each layer examples. I thought the examples are to explain the behavior of the module.

File: docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md [hunk 24]
```diff
@@ -0,0 +1,77 @@
+## SelectTable ##
+
+**Scala:**
+```scala
+SelectTable(index: Int)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md]: use
`val m = ...`

File: docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+```
+**Python:**
+```python
+SelectTable(dimension)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Merge Layers/SelectTable.md]: use
`m=SelectTable(dimension)`

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/ELU.md [hunk 26]
```diff
@@ -0,0 +1,81 @@
+## ELU ##
+
+
+**Scala:**
+
+```scala
+ELU(alpha = 1.0, inplace = false)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/ELU.md]: use val m = ...

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/ELU.md [hunk 27]
```diff
@@ -0,0 +1,81 @@
...
+```
+**Python:**
+```python
+ELU(alpha=1.0, inplace=False)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/ELU.md]: use val m = ...

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/HardShrink.md [hunk 28]
```diff
@@ -0,0 +1,83 @@
+## HardShrink ##
+
+
+**Scala:**
+```scala
+HardShrink(lambda = 0.5)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/HardShrink.md]: use val m = ...

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/HardShrink.md [hunk 29]
```diff
@@ -0,0 +1,83 @@
...
+```
+**Python:**
+```python
+HardShrink(the_lambda=0.5)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/HardShrink.md]: use val m = ...

File: docs/docs/APIdocs/layers/nn/Simple Layers/Abs.md [hunk 30]
```diff
@@ -0,0 +1,37 @@
+## Abs ##
+
+**Scala:**
+```scala
+abs = Abs()
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Abs.md]: use val m = ...

File: docs/docs/APIdocs/layers/nn/Simple Layers/Abs.md [hunk 31]
```diff
@@ -0,0 +1,37 @@
...
+```
+**Python:**
+```python
+abs = Abs()
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Abs.md]: use val m = ...

**Pull request merged**